src/pages/instances/actions/snapshots/ActionButtons.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, useState } from "react";
+import { LxdInstance, LxdSnapshot } from "types/instance";
+import { deleteSnapshot, restoreSnapshot } from "api/snapshots";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import ConfirmationButton from "components/ConfirmationButton";
+import { List } from "@canonical/react-components";
+import classnames from "classnames";
+
+interface Props {
+  instance: LxdInstance;
+  snapshot: LxdSnapshot;
+  onSuccess: (message: string) => void;
+  onFailure: (message: string, e: unknown) => void;
+}
+
+const ActionButtons: FC<Props> = ({
+  instance,
+  snapshot,
+  onSuccess,
+  onFailure,
+}) => {
+  const [isLoading, setLoading] = useState(false);
+  const queryClient = useQueryClient();
+
+  const handleDelete = () => {
+    setLoading(true);
+    deleteSnapshot(instance, snapshot)
+      .then(() => {
+        setLoading(false);
```

Comment:
I started to put the `setLoading` and `invalidateQueries` into a [finally block](https://github.com/canonical/lxd-ui/pull/231/files#diff-e7ef45c9e95848b69d5608e22eaae94b539591811aa5b705ded87d29291f9a92R41). I think it makes the code a bit more structured. And we might want to refresh state even in case an error happened.

Reply:
Very good idea! I applied it to both action handlers.